# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build gst-plugins-bad-build gst-plugins-good-build gst-plugins-ugly-build"

PKG_CONFIG_PATH=/wl/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal -I common/m4; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/wl/lib/pkgconfig LDFLAGS=-Wl,-rpath,/wl/lib ./configure --prefix=/wl

# build
make

# install
install -d $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-address-pool.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-auth.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-client.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-context.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-media.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-media-factory.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-media-factory-uri.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-mount-points.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-params.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-permissions.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-sdp.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-server.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-server-object.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-server-prelude.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-session.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-session-media.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-session-pool.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-stream.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-stream-transport.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-thread-pool.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -m 644 gst/rtsp-server/rtsp-token.h $DESTDIR/wl/include/gstreamer-1.0/gst/rtsp-server
install -d $DESTDIR/wl/lib
install gst/rtsp-server/.libs/libgstrtspserver-1.0.so.0.* $DESTDIR/wl/lib/libgstrtspserver-1.0.so.0
ln -sf libgstrtspserver-1.0.so.0 $DESTDIR/wl/lib/libgstrtspserver-1.0.so
install -d $DESTDIR/wl/lib/gstreamer-1.0
install gst/rtsp-sink/.libs/libgstrtspclientsink.so $DESTDIR/wl/lib/gstreamer-1.0
install -d $DESTDIR/wl/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-rtsp-server-1.0.pc $DESTDIR/wl/lib/pkgconfig

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/gst-rtsp-server-build.pc << EOF
Name: gst-rtsp-server
Version: 1.14.5
Description: GStreamer RTSP server library
Requires: $REQUIRES

devel=\\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-address-pool.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-auth.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-client.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-context.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-media.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-media-factory.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-media-factory-uri.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-mount-points.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-params.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-permissions.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-sdp.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-server.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-server-object.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-server-prelude \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-session.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-session-media.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-session-pool.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-stream.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-stream-transport.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-thread-pool.h \\
/wl/include/gstreamer-1.0/gst/rtsp-server/rtsp-token.h \\
/wl/lib/libgstrtspserver-1.0.so \\
/wl/lib/pkgconfig/gstreamer-rtsp-server-1.0.pc

exec=\\
/wl/lib/libgstrtspserver-1.0.so.0 \\
/wl/lib/gstreamer-1.0/libgstrtspclientsink.so
EOF

exit
--- gst-rtsp-server-1.14.5.orig/examples/test-netclock-client.c
+++ gst-rtsp-server-1.14.5/examples/test-netclock-client.c
@@ -21,7 +21,7 @@
 #include <stdlib.h>
 
 #include <gst/gst.h>
-#include <gst/net/gstnet.h>
+#include <gst/net/net.h>
 
 #define PLAYBACK_DELAY_MS 40
 
--- gst-rtsp-server-1.14.5.orig/gst/rtsp-server/rtsp-media.h
+++ gst-rtsp-server-1.14.5/gst/rtsp-server/rtsp-media.h
@@ -19,7 +19,7 @@
 
 #include <gst/gst.h>
 #include <gst/rtsp/rtsp.h>
-#include <gst/net/gstnet.h>
+#include <gst/net/net.h>
 
 #ifndef __GST_RTSP_MEDIA_H__
 #define __GST_RTSP_MEDIA_H__
--- gst-rtsp-server-1.14.5.orig/gst/rtsp-server/rtsp-server.h
+++ gst-rtsp-server-1.14.5/gst/rtsp-server/rtsp-server.h
@@ -46,11 +46,6 @@
 #include "rtsp-media-factory-uri.h"
 #include "rtsp-params.h"
 
-#include "rtsp-onvif-client.h"
-#include "rtsp-onvif-media-factory.h"
-#include "rtsp-onvif-media.h"
-#include "rtsp-onvif-server.h"
-
 G_END_DECLS
 
 #endif /* __GST_RTSP_SERVER_H__ */
--- gst-rtsp-server-1.14.5.orig/gst/rtsp-sink/gstrtspclientsink.c
+++ gst-rtsp-server-1.14.5/gst/rtsp-sink/gstrtspclientsink.c
@@ -95,7 +95,7 @@
 #include <stdio.h>
 #include <stdarg.h>
 
-#include <gst/net/gstnet.h>
+#include <gst/net/net.h>
 #include <gst/sdp/gstsdpmessage.h>
 #include <gst/sdp/gstmikey.h>
 #include <gst/rtp/rtp.h>
